cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -284,10 +284,11 @@ func (m *asgCache) DeleteInstances(instances []*AwsInstanceRef) error {
 			}
 
 			if lifecycle != nil &&
+				*lifecycle == autoscaling.LifecycleStateTerminated ||
```

Comment:
this one is the last of the terminating states, so might just include it too
https://github.com/aws/aws-sdk-go/blob/v1.45.19/service/autoscaling/api.go#L21814-L21824